server/controllers/notifications.js

```diff
@@ -0,0 +1,98 @@
+const Notification = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+
+// @desc    Get all Notifications
+// @route   GET /notifications
+// @access  Private
+exports.getNotifications = asyncHandler(async (req, res) => {
+  const { id } = req.user;
+  const userNotification = await Notification.findOne({ receiver: id });
+  if (userNotification && userNotification.notifications.length > 0) {
+    return res.status(200).json({
+      result: userNotification.notifications.length,
+      data: userNotification.notifications,
+    });
+  } else {
+    res.status(400);
+    throw new Error("No notifications found");
+  }
+});
+
+// @desc    Get Unread Notifications
+// @route   GET /unreadnotifications
+// @access  Private
+exports.getUnReadNotifications = asyncHandler(async (req, res) => {
+  const { id } = req.user;
+  const userNotification = await Notification.findOne({ receiver: id });
+  if (userNotification && userNotification.notifications.length > 0) {
+    const unReadNotifications = userNotification.notifications.filter(
+      (notification) => notification.isRead === false
+    );
+    return res.status(200).json({
+      result: unReadNotifications.length,
+      data: unReadNotifications,
+    });
+  } else {
+    res.status(400);
+    throw new Error("No unread notifications found");
+  }
+});
+
+// @desc    Create a new notification
+// @route   POST /notifications
+// @access  Private
+exports.createNotification = asyncHandler(async (req, res) => {
+  const { id } = req.user;
+  const { senderId } = req.params;
+  const { title, message, type } = req.body;
+  const userNotification = await Notification.findOne({ receiver: id });
+
+  if (!userNotification) {
+    await new Notification({
+      receiver: id,
+      notifications: [
+        {
+          type,
+          sender: senderId,
+          title,
+          message,
+        },
+      ],
+    }).save();
```

Comment:
To be more readable, it's good to write this on a separate line

Reply:
Prettier does not let me, not my fault 😁